src/App.tsx

```diff
@@ -30,16 +32,21 @@ class App extends Component {
   handleLogout = async () => {
     await Auth.signOut();
     this.props.history.push('/login');
+    this.props.userHasAuthenticated(false);
   }
 
   render() {
     return (
       !this.state.isAuthenticating &&
-      <div className="App container">
-        <ScreensRoot />
-      </div>
+        <div className="App container">
+          <ScreensRoot />
+        </div>
     );
   }
 }
 
-export default withRouter(App);
+const mapDispatchToProps = dispatch => ({
```

Comment:
Looking good.
Moreover, you can take a look at this method `bindActionCreators` (https://redux.js.org/api/bindactioncreators). Maybe it can help to make this method better 😄